test/test.py

```diff
@@ -876,6 +878,20 @@ def test_multiple_requests_with_illegal_traceparent(self):
 		self.assertFalse('00000000000000000000000000000000' in trace_ids)
 		self.assertEqual(len(parent_ids), 3)
 
+@unittest.skipIf(SPEC_LEVEL < 2, "Trace Context Level 2")
+class TraceContext2Test(TestBase):
+	def test_propagates_random_flag(self):
+		'''
+		harness sends a request with a traceparent with the random flag set
+		expects a valid traceparent from the output header, with the random flag set
+		'''
+		traceparent, tracestate = self.make_single_request_and_get_tracecontext([
+			['traceparent', '00-12345678901234567890123456789012-1234567890123456-02'],
+		])
+		self.assertEqual(traceparent.trace_id.hex(), '12345678901234567890123456789012')
+		self.assertNotEqual(traceparent.parent_id.hex(), '1234567890123456')
```

Comment:
Is the check for parent ID meant to be an "assertEqual" check instead of "assertNotEqual"?  Looking at the parent ID being compared here, it seems to match the parentID specified in line 889.

Reply:
The request from the service being tested should have a new parent ID, not reuse the incoming one

Reply:
Trace ID should be equal and Trace Parent should be not equal. This test is correct. See a similar example here https://github.com/w3c/trace-context/blob/84b583d86ecb7005a9eab8fed86ab7117b050b48/test/test.py#L123-L132

Reply:
Got it, thanks for the clarification.